tests/tensorizer_loader/test_tensorizer.py

```diff
@@ -280,3 +292,196 @@ def test_vllm_tensorized_model_has_same_outputs(vllm_runner, tmp_path):
+                              serialization_kwargs=serialization_params)
+
+    args = EngineArgs(model=model_ref, device="cuda")
+    tensorize_vllm_model(args, config)
+
+    stream_kwargs = {
+        "mode": "foo"
+    }
+
+    loader_tc = TensorizerConfig(
+        tensorizer_uri=str(model_path),
+        deserialization_kwargs=deserialization_params,
+        stream_kwargs=stream_kwargs,
+    )
+
+    # Like in the last test, purposefully breaking the loading process
+    # by passing an illegal value for the "mode" parameter for open_stream.
+    # If the ValueError we're looking for is produced, we've confirmed that
+    # we're able to pass stream_kwargs to configure the stream
+    try:
+        LLM(
+            model=model_ref,
+            load_format="tensorizer",
+            model_loader_extra_config=loader_tc,
+        )
+    except RuntimeError:
+        out, err = capfd.readouterr()
+        combined_output = out + err
+        assert ("ValueError: Only binary modes (\"rb\", \"wb\", \"wb+\", etc.) are valid when opening local file streams.") in combined_output
+
+@pytest.mark.asyncio
+async def test_serialize_and_serve_entrypoints(tmp_path):
+    model_ref = "facebook/opt-125m"
+
+    suffix = "test"
+    try:
+        result = subprocess.run([
+            sys.executable,
+            f"{VLLM_PATH}/examples/others/tensorize_vllm_model.py", "--model",
+            model_ref, "serialize", "--serialized-directory",
+            str(tmp_path), "--suffix", suffix, "--serialization-kwargs",
+            json.dumps({"limit_cpu_concurrency": 4})
+        ],
+                                check=True,
+                                capture_output=True,
+                                text=True)
+    except subprocess.CalledProcessError as e:
+        print("Tensorizing failed.")
+        print("STDOUT:\n", e.stdout)
+        print("STDERR:\n", e.stderr)
+        raise
+
+    print("Serializing STDOUT:\n", result.stdout)
+
+    # Next, try to serve with vllm serve
+    model_uri = tmp_path / "vllm" / model_ref / suffix / "model.tensors"
+
+    model_loader_extra_config = {
+        "tensorizer_uri": str(model_uri),
+        "stream_kwargs": {
+            "force_http": False,
+        },
+        "deserialization_kwargs": {
+            "verify_hash": True,
+            "num_readers": 8,
+        }
+    }
+
+    cmd = ["-m", "vllm.entrypoints.cli.main", "serve", "--host", "localhost", "--load-format", "tensorizer", model_ref, "--model-loader-extra-config", json.dumps(model_loader_extra_config, indent=2)]
+
+    proc = await asyncio.create_subprocess_exec(
+        sys.executable,
+        *cmd,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+    )
+
+
+    try:
+        async with asyncio.timeout(180):
+            await proc.stdout.readuntil(b"Application startup complete.")
+
+    except asyncio.TimeoutError:
+        pytest.fail("Server did not start successfully")
+
+
+    proc.terminate()
```

Comment:
Use `finally` to ensure the server is closed instead of letting it potentially hang indefinitely in a subprocess
```suggestion
    try:
        async with asyncio.timeout(180):
            await proc.stdout.readuntil(b"Application startup complete.")
    except asyncio.TimeoutError:
        pytest.fail("Server did not start successfully")
    finally:
        proc.terminate()
```